src/lib/operators/join_hash.cpp

```diff
@@ -64,6 +64,53 @@ std::shared_ptr<AbstractOperator> JoinHash::_on_deep_copy(
 
 void JoinHash::_on_set_parameters(const std::unordered_map<ParameterID, AllTypeVariant>& parameters) {}
 
+template <typename T>
+size_t JoinHash::_calculate_radix_bits(const std::shared_ptr<const Table> build_table,
```

Comment:
Const method?

Reply:
Changed to public static with tests.

Reply:
Sorry, forgot to WIP it.

But I think it's final now.